relay-cardinality/src/redis/quota.rs

```diff
@@ -64,6 +69,25 @@ impl QuotaScoping {
         UnixTimestamp::from_secs(timestamp.as_secs() + shift)
     }
 
+    pub fn full(self, entry: Entry<'_>) -> QuotaScoping {
+        let name = match self.scope {
+            CardinalityScope::Name => Some(fnv32(entry.name)),
+            _ => None,
+        };
+
+        QuotaScoping { parent: self, name }
+    }
+}
+
+/// A quota scoping extracted from a [`CardinalityLimit`], a [`Scoping`]
+/// and a [`CardinalityItem`](crate::CardinalityItem).
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct QuotaScoping {
+    parent: PartialQuotaScoping,
+    name: Option<u32>,
```

Comment:
Haven't reviewed the PR fully, but what makes the `name` special, i.e. more special than the namespace? Why is the namespace part of `PartialQuotaScoping` but `name` is not?